### PR TITLE
Add optional job_project_id config to BigQuery crawlers

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

To allow running BigQuery jobs using a project ID different from the default.

### 🤓 What?

As titled.

### 🧪 Tested?

Manually tested against a production deployment.

### CODE REVIEW

File: metaphor/bigquery/profile/extractor.py [hunk 1]
```diff
@@ -55,6 +56,7 @@ async def extract(
 
         logger.info("Fetching usage info from BigQuery")
 
+        self._job_project_id = config.project_id
```
Comment (alyiwang) [inline metaphor/bigquery/profile/extractor.py:59]: Should use `config.job_project_id` and fallback to `config.project_id`?

File: metaphor/bigquery/extractor.py [hunk 2]
```diff
@@ -88,7 +88,8 @@ def get_table(table: bigquery.TableReference) -> Dataset:
 
             logger.info(f"Getting table DDL for {dataset_ref}")
             table_ddl = client.query(
-                f"select table_name, ddl from `{dataset_ref.project}.{dataset_ref.dataset_id}.INFORMATION_SCHEMA.TABLES`"
+                f"select table_name, ddl from `{dataset_ref.project}.{dataset_ref.dataset_id}.INFORMATION_SCHEMA.TABLES`",
+                project=config.job_project_id,
```
Comment (alyiwang) [inline metaphor/bigquery/extractor.py:92]: Should use `config.job_project_id` and fallback to `config.project_id`?

Comment (mars-lan) [reply metaphor/bigquery/extractor.py:92]: If `job_project_id` is None,`client` will use the project ID it's created with, which is `config.project_id` or the credential's project ID: https://github.com/MetaphorData/connectors/blob/main/metaphor/bigquery/utils.py#L31

File: metaphor/bigquery/profile/extractor.py (hunk 1 above)
Comment (alyiwang) [inline metaphor/bigquery/profile/extractor.py:59]:
```suggestion
        self._job_project_id = config.job_project_id
```

Comment (mars-lan) [reply metaphor/bigquery/profile/extractor.py]: Thanks for catching.

File: metaphor/bigquery/extractor.py (hunk 2 above)
Comment (alyiwang) [reply metaphor/bigquery/extractor.py:92]: I see 👍

**Review by alyiwang — APPROVED** [review]
LGTM

**Pull request merged**